build_node/build_node_supervisor.py

```diff
@@ -47,18 +98,12 @@ def run(self):
             if not any(builders_aliveness):
                 logging.warning('All builders are dead, exiting')
                 break
-            active_tasks = self.get_active_tasks()
-            logging.debug('Sending active tasks: {}'.format(active_tasks))
-            full_url = urllib.parse.urljoin(
-                self.config.master_url, 'build_node/ping'
-            )
-            data = {'active_tasks': [int(item) for item in active_tasks]}
-            try:
-                self.__session.post(
-                    full_url, json=data, timeout=self.config.request_timeout)
-            except Exception:
-                logging.error(
-                    "Can't report active task to master:\n%s",
-                    traceback.format_exc()
-                )
-            self.terminated_event.wait(60)
+            self.__report_active_tasks()
+            task = self.__request_build_task()
```

Comment:
If implemented the thing from above, please put check for queue being full here and only call for new task when there are empty slots